surface_potential_analysis/kernel/build.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable, TypeVar
+
+from scipy.constants import Boltzmann
+
+from surface_potential_analysis.basis.util import get_displacements_x
+from surface_potential_analysis.kernel.kernel import (
+    SingleBasisDiagonalNoiseOperatorList,
+    SingleBasisNoiseOperatorList,
+    as_noise_operators,
+)
+from surface_potential_analysis.operator.operations import (
+    add_list_list,
+    get_commutator_operator_list,
+    scale_operator_list,
+)
+from surface_potential_analysis.operator.operator import SingleBasisOperator
+from surface_potential_analysis.stacked_basis.conversion import (
+    stacked_basis_as_fundamental_position_basis,
+)
+
+if TYPE_CHECKING:
+    import numpy as np
+
+    from surface_potential_analysis.basis.basis import FundamentalPositionBasis
+    from surface_potential_analysis.basis.basis_like import BasisLike
+    from surface_potential_analysis.basis.stacked_basis import (
+        StackedBasisWithVolumeLike,
+        TupleBasisWithLengthLike,
+    )
+    from surface_potential_analysis.kernel.kernel import (
+        IsotropicNoiseKernel,
+        SingleBasisNoiseOperatorList,
+    )
+    from surface_potential_analysis.operator.operator import SingleBasisOperator
+    from surface_potential_analysis.operator.operator_list import (
+        SingleBasisOperatorList,
+    )
+
+    _B1 = TypeVar(
+        "_B1",
+        bound=BasisLike[Any, Any],
+    )
+
+    _B0 = TypeVar(
+        "_B0",
+        bound=BasisLike[Any, Any],
+    )
+    _B2 = TypeVar(
+        "_B2",
+        bound=BasisLike[Any, Any],
+    )
+
+
+def build_isotropic_kernel_from_function(
+    basis: StackedBasisWithVolumeLike[Any, Any, Any],
+    fn: Callable[
+        [np.ndarray[Any, np.dtype[np.float64]]],
+        np.ndarray[Any, np.dtype[np.complex128]],
+    ],
+) -> IsotropicNoiseKernel[
+    TupleBasisWithLengthLike[*tuple[FundamentalPositionBasis[Any, Any], ...]],
+]:
+    """
+    Get an Isotropic Kernel with a correllation beta(x-x').
+
+    Parameters
+    ----------
+    basis : StackedBasisWithVolumeLike[Any, Any, Any]
+    fn : Callable[
+        [np.ndarray[Any, np.dtype[np.float64]]],
+        np.ndarray[Any, np.dtype[np.complex128]],
+    ]
+        beta(x-x'), the correllation as a function of displacement
+
+    Returns
+    -------
+    IsotropicNoiseKernel[
+        TupleBasisWithLengthLike[*tuple[FundamentalPositionBasis[Any, Any], ...]],
+    ]
+    """
+    displacements = get_displacements_x(basis)[0]
+    correlation = fn(displacements)
+
+    basis_x = stacked_basis_as_fundamental_position_basis(basis)
```

Comment:
why not get displacements from fundamental position basis but just from basis?

Reply:
Good point - I modified get_displacements to return a ValueList, and then use that basis for the kernel